vnext/Desktop/FeatureGate.cpp

```diff
@@ -0,0 +1,21 @@
+#include <FeatureGate.h>
+
+#include <unordered_set>
+
+using std::string;
+
+namespace {
+std::unordered_set<string> g_featureGates;
+}
+
+namespace Microsoft::React {
+
+void SetFeatureGate(string &&name) noexcept {
+  g_featureGates.insert(std::move(name));
+}
+
+bool GetFeatureGate(string &&name) noexcept {
+  return g_featureGates.find(std::move(name)) != g_featureGates.end();
```

Comment:
>find [](start = 24, length = 4)

(nit) this can take const&, right?

Reply:
Agree. The name here should be 'string const&'. I wish we could use string_view instead, but it would only work today for std::map and std::set. [https://stackoverflow.com/questions/49709548/c-unordered-mapstring-lookup-without-constructing-string](https://stackoverflow.com/questions/49709548/c-unordered-mapstring-lookup-without-constructing-string)

---
In reply to: [451661713](https://github.com/microsoft/react-native-windows/pull/5458#discussion_r451661713) [](ancestors = 451661713)

Reply:
Not following. What is the advantage of string `const&` over `string&&`?
Note, in all likelyhood, these names will be provided as string literals and not as variables.

Reply:
&& should be used in Constructors and Setters, where we know that there will be an assignment, and we want to avoid object copying in this assignment.

Here, you are just retrieving something, so no need for assignment, thus no reason for avoiding copying (because there will be no copy), thus const& is better.

The advantage of string& over string&& for non-assignment Apis is that if caller wants to continue accessing string after calling this Api, caller doesn't have to create a additional local copy.

It's a short explanation, but let's chat if more clarification is needed.

---
In reply to: [451924382](https://github.com/microsoft/react-native-windows/pull/5458#discussion_r451924382) [](ancestors = 451924382)

Reply:
This explanation makes sense. At first, I understood we should switch to `const string&` on both functions, but doing it only for the getter is clear enough.